beacon-chain/sync/rpc_data_column_sidecars_by_root.go

```diff
@@ -0,0 +1,182 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"slices"
+	"time"
+
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/p2p/types"
+	"github.com/OffchainLabs/prysm/v6/cmd/beacon-chain/flags"
+	fieldparams "github.com/OffchainLabs/prysm/v6/config/fieldparams"
+	"github.com/OffchainLabs/prysm/v6/config/params"
+	"github.com/OffchainLabs/prysm/v6/consensus-types/primitives"
+	"github.com/OffchainLabs/prysm/v6/monitoring/tracing"
+	"github.com/OffchainLabs/prysm/v6/monitoring/tracing/trace"
+	"github.com/OffchainLabs/prysm/v6/time/slots"
+	libp2pcore "github.com/libp2p/go-libp2p/core"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var notDataColumnsByRootIdentifiersError = errors.New("not data columns by root identifiers")
+
+// dataColumnSidecarByRootRPCHandler handles the data column sidecars by root RPC request.
+// https://github.com/ethereum/consensus-specs/blob/dev/specs/fulu/p2p-interface.md#datacolumnsidecarsbyroot-v1
+func (s *Service) dataColumnSidecarByRootRPCHandler(ctx context.Context, msg interface{}, stream libp2pcore.Stream) error {
+	ctx, span := trace.StartSpan(ctx, "sync.dataColumnSidecarByRootRPCHandler")
+	defer span.End()
+
+	batchSize := flags.Get().DataColumnBatchLimit
+	numberOfColumns := params.BeaconConfig().NumberOfColumns
+
+	// Check if the message type is the one expected.
+	ref, ok := msg.(*types.DataColumnsByRootIdentifiers)
+	if !ok {
+		return notDataColumnsByRootIdentifiersError
+	}
+
+	requestedColumnIdents := *ref
+	remotePeerId := stream.Conn().RemotePeer()
+
+	ctx, cancel := context.WithTimeout(ctx, ttfbTimeout)
```

Comment:
where is ttfbTimeout defined

Reply:
https://github.com/OffchainLabs/prysm/blob/cd646dc32f77404eb7d9a915746af87f279dc48e/beacon-chain/sync/rpc.go#L29